cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -197,7 +205,7 @@ struct GrouperFastImpl : Grouper {
 #if ARROW_LITTLE_ENDIAN
     for (size_t i = 0; i < keys.size(); ++i) {
       const auto& key = keys[i].type;
-      if (is_large_binary_like(key->id())) {
+      if (is_large_binary_like(key->id()) || key->id() == Type::NA) {
```

Comment:
I'm not sure I understand, why cannot null keys be used here?

Reply:
In the previous commits, GrouperFastImpl for null type is not implemented yet, so null should be skipped for GrouperFastImpl. Now the null type is added to GrouperFastImpl, so the condition is removed.